### DISCUSSION TITLE
How to access Zustand persist API?

### DISCUSSION THREAD

**Discussion by vnugent** [udecode/zustand-x#45] [Q&A] [answered]
I want to manually call  Zustand `persist.rehydrate()` as documented [here](https://github.com/pmndrs/zustand/blob/main/docs/persisting-store-data.md#api).

```
await userCollectionsStore.persist.rehydrate() // persist is not defined
```

** Edited
Found it in `useStore()`.  Now how do I call `rehydrate()` in `useEffect()`?

> Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:

```javascript
useEffect(()=>{
  // do some check to see if we need to rehydrate store
  // @ts-expect-error
  void userCollectionsStore.useStore().persist.rehydrate()  // Error: Invalid hook call. 
})
```

**Comment by zbeyens** [comment, marked as answer]
The hook needs to be called outside `useEffect`, can you try this:

```tsx
const store = userCollectionsStore.useStore()

useEffect(()=>{
  // do some check to see if we need to rehydrate store
  // @ts-expect-error
  store.persist.rehydrate()  // Error: Invalid hook call. 
})
```
